src/test/bufferlist.cc

```diff
@@ -2269,6 +2269,120 @@ TEST(BufferList, append_zero) {
   EXPECT_EQ('\0', bl[1]);
 }
 
+TEST(BufferList, appender_bench) {
+  unsigned long long ma = 1024 * 16, mb = 65536;
+  cout << "appending " << (ma * mb) << " bytes" << std::endl;
+  {
+    utime_t start = ceph_clock_now(NULL);
+    uint64_t v = 123;
+    for (unsigned a = 0; a < ma; ++a) {
+      bufferlist bl;
+      for (unsigned b = 0; b < mb; ++b) {
+	bl.append((char*)&v, sizeof(v));
+      }
+    }
+    utime_t dur = ceph_clock_now(NULL) - start;
+    cout << "buffer::list::append " << dur << std::endl;
+  }
+  {
+    utime_t start = ceph_clock_now(NULL);
+    uint64_t v = 123;
+    for (unsigned a = 0; a < ma; ++a) {
+      bufferlist bl;
+      for (unsigned b = 0; b < mb; ++b) {
+	::encode(v, bl);
+      }
+    }
+    utime_t dur = ceph_clock_now(NULL) - start;
+    cout << "buffer::list encode " << dur << std::endl;
+  }
+
+  {
+    utime_t start = ceph_clock_now(NULL);
+    uint64_t v = 123;
+    for (unsigned a = 0; a < ma; ++a) {
+      bufferlist bl;
+      bufferlist::safe_appender ap = bl.get_safe_appender();
+      for (unsigned b = 0; b < mb; ++b) {
+	ap.append((char*)&v, sizeof(v));
+      }
+    }
+    utime_t dur = ceph_clock_now(NULL) - start;
+    cout << "buffer::list::safe_appender::append " << dur << std::endl;
+  }
+  {
+    utime_t start = ceph_clock_now(NULL);
+    uint64_t v = 123;
+    for (unsigned a = 0; a < ma; ++a) {
+      bufferlist bl;
+      bufferlist::safe_appender ap = bl.get_safe_appender();
+      for (unsigned b = 0; b < mb; ++b) {
+	ap.append_v(v);
+      }
+    }
+    utime_t dur = ceph_clock_now(NULL) - start;
+    cout << "buffer::list::safe_appender::append_v " << dur << std::endl;
+  }
+  {
+    utime_t start = ceph_clock_now(NULL);
+    uint64_t v = 123;
+    for (unsigned a = 0; a < ma; ++a) {
+      bufferlist bl;
+      bufferlist::safe_appender ap = bl.get_safe_appender();
+      for (unsigned b = 0; b < mb; ++b) {
+	::encode(v, ap);
+      }
+    }
+    utime_t dur = ceph_clock_now(NULL) - start;
+    cout << "buffer::list::safe_appender encode " << dur << std::endl;
+  }
+
+  {
+    utime_t start = ceph_clock_now(NULL);
+    uint64_t v = 123;
+    for (unsigned a = 0; a < ma; ++a) {
+      bufferlist bl;
+      bufferlist::unsafe_appender ap = bl.get_unsafe_appender(sizeof(v) * mb);
```

Comment:
You pre-allocate 4K for safe allocator and 64K*8 for unsafe one. Hence your comparison for safe vs.  unsafe isn't pretty adequate. Suggest to make similar values and compare. Probably the diff between them wouldn't be that much.